Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -996,6 +1007,104 @@ bool parse_variable_attachments(Iterator &first, Iterator last,
   }
   return true;
 }
+
+template <typename Iterator>
+bool parse_wedged_bonds(Iterator &first, Iterator last, RDKit::RWMol &mol,
+                        unsigned int startAtomIdx, unsigned int startBondIdx) {
+  // these look like: CC(O)Cl |w:1.0|
+  // also wD and wU for down and up wedges.
+  //
+  // We do not end up using this to set stereochemistry, but the relevant bond
+  // properties are set in case client code wants to do something with the
+  // information.
+  if (first >= last || *first != 'w' || first + 1 >= last) {
+    return false;
+  }
+  ++first;
+  Bond::BondDir state = Bond::BondDir::NONE;
+  unsigned int cfg = 0;
+  switch (*first) {
+    case ':':
+      state = Bond::BondDir::UNKNOWN;
+      cfg = 2;
+      break;
+    case 'U':
+      state = Bond::BondDir::BEGINWEDGE;
+      cfg = 1;
+      ++first;
+      break;
+    case 'D':
+      state = Bond::BondDir::BEGINDASH;
+      cfg = 3;
+      ++first;
+      break;
+    default:
+      break;
+  }
+  if (state == Bond::BondDir::NONE || first >= last || first + 1 >= last ||
+      *first != ':') {
+    return false;
+  }
+  ++first;
+  while (first < last && *first >= '0' && *first <= '9') {
+    unsigned int atomIdx;
+    if (!read_int(first, last, atomIdx)) {
+      return false;
+    }
+    if (first < last && *first == '.') {
+      ++first;
+    } else {
+      BOOST_LOG(rdWarningLog) << "improperly formatted w block" << std::endl;
+      return false;
+    }
+    unsigned int bondIdx;
+    if (!read_int(first, last, bondIdx)) {
+      return false;
+    }
+
+    if (!VALID_ATIDX(atomIdx)) {
+      BOOST_LOG(rdWarningLog)
+          << "bad atom index, " << atomIdx << ", in w block" << std::endl;
+      return false;
+    }
+    if (!VALID_BNDIDX(bondIdx)) {
+      BOOST_LOG(rdWarningLog)
+          << "bad bond index, " << bondIdx << ", in w block" << std::endl;
+      return false;
+    }
+    auto atom = mol.getAtomWithIdx(atomIdx + startAtomIdx);
+    auto bond = mol.getBondWithIdx(bondIdx + startBondIdx);
+
+    // first things first, the atom needs to be the start atom of the bond for
+    // any of this to make sense
+    if (atom->getIdx() != bond->getBeginAtomIdx()) {
+      if (atom->getIdx() != bond->getEndAtomIdx()) {
+        BOOST_LOG(rdWarningLog)
+            << "atom " << atomIdx << " is not associated with bond " << bondIdx
+            << " in w block" << std::endl;
+        return false;
+      }
+      auto eidx = bond->getBeginAtomIdx();
```

Comment:
> We swap the bond begin/end here. Is there anything else that depends on the begin vs end? How do we know that we're not breaking that?

Dative (=coordination) bonds have an order dependence.
We don't know that we aren't breaking something else here... we're just accepting what's in the CXSMILES as correct and not engaging in defensive programming. It would be worth thinking about adding some more consistency checks in CXSMILES parser in general, but I don't think that's for here.